skema/program_analysis/CAST2FN/ann_cast/to_gromet_pass.py

```diff
@@ -930,7 +924,6 @@ def determine_func_type(self, node):
                     # TODO: import_version/import_source
                     return (FunctionType.IMPORTED, ImportType.OTHER, None, None, "Python", "3.10")
```

Comment:
I see 3.10 here.  Can you explain how this relates (or doesn't relate) to skema being stuck on 3.8?

Reply:
I'll let @titomeister answer for confirmation, but I think in this case the 3.10 is referring to the target source language version, not the source language of skema. Though I'm curious how that function would behave if passed Fortran code instead of Python.

Reply:
I think I just assumed that any code written nowadays would be Python 3.10 (or whatever the latest version is now as that was a little while ago). It doesn't have any real relation to us being stuck on 3.8.
The reason we're stuck on 3.8 has to do when we ingest a file and generate its Python AST, the Python AST that gets generated has changed from version 3.8 and the handlers aren't yet setup correctly to handle the AST from later versions of Python. If I recall correctly it's only the handler that has to do with indexing that needs some changes.

I'm not sure how the function would behave with Fortran code, as I think I wrote this before we started supporting modules in Fortran. It probably needs some changes to appropriately generate the correct Gromet for Fortran.